Microsoft.Toolkit.Win32.UI.Controls/Interop/WinRT/WindowsRuntimeCollection.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT
+{
+    public sealed class WindowsRuntimeCollection<T1, T2> : IList<T1>
+    {
+        private readonly IList<T2> collection;
+        private readonly Func<T2, T1> converter1;
+        private readonly Func<T1, T2> converter2;
+
+        public WindowsRuntimeCollection(IList<T2> collection, Func<T2, T1> converter1, Func<T1, T2> converter2)
+        {
+            this.collection = collection;
+            this.converter1 = converter1;
+            this.converter2 = converter2;
+        }
+
+        T1 IList<T1>.this[int index]
```

Comment:
Why are all the interfaces explicitly implemented?  That seems to make this type harder to use as callers need to cast it to the right interface?